mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/NotificationInfo.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+import static org.mule.runtime.api.notification.AbstractServerNotification.TYPE_INFO;
+
+/**
+ * Holds all notification related information that will be used to fire a desired notification.
+ *
+ * @param <T> the type of data provided
+ * @since 1.1
+ */
+public interface NotificationInfo<T> {
```

Comment:
what is the purpose of the generic? You will never be able to use it when handling the notification